datalake/backend/tests/collection.json

```diff
@@ -0,0 +1,104 @@
+{
+    "id": "test_dataset",
+    "stac_version": "1.0.0-beta.2",
+    "description": "This is dataset description.",
+    "links": [
+        {
+            "rel": "item",
+            "href": "s3://s3-bucket-name/datasets/xyz/1234.json",
+            "type": "application/json"
+        },
+        {
+            "rel": "root",
+            "href": "s3://s3-bucket-name/catalog.json",
```

Comment:
`s3-bucket-name` is going to fail validation, so we probably want to use `f"linz-geospatial-data-lake-{ENV}"` here as well.

I would also use `any_safe_filename()` to clarify that the filename doesn't matter for the test.

Reply:
The problem with using `linz-geospatial-data-lake-{ENV}` S3 bucket is fact that this is Data Lake Storage bucket (destination). We should use some other bucket as source bucket.

Reply:
Probably I should create new stack containing S3 bucket for input test datasets which we deploy only for testing

Reply:
In 3f81332ad4d2e2c7f3da46537381b1e04190a70d , I created dataset source stack containing separate S3 bucket for source datasets. We are going to need it anyway for customers which are not able to host source S3 bucket by them selves (this was discussed many times already).